xbmc/Application.cpp

```diff
@@ -1211,6 +1211,9 @@ bool CApplication::Initialize()
   CServiceBroker::GetRepositoryUpdater().Start();
   CServiceBroker::GetServiceAddons().Start();
 
+  CAddonMgr::GetInstance().Events().Subscribe(this, &CApplication::OnAddonEvent);
```

Comment:
A `CServiceBroker::GetAddonMgr().Events().Unsubscribe(this);` is missing.

Reply:
Can `Subscribe` be done from CAdvancedSettings itself?

Reply:
As is init before app is init, tried before, so no.

Reply:
We could expose a method that do subscribe/unsubscribe but need to call it in `CApplication` let me know what you prefer.

Reply:
I prefer this to be removed from both CApplication and CAdvancedSettings, and rather made a separate 'service' that holds the extension info. It can then read extensions from both AS and addons. Others that needs the combined list reads from it. 

URIUtils::RemoveExtension should be moved here too to get rid of the dependency on globals.

Reply:
This will end up in a massive refactor because we use AS's `GetXXXExtensions` everywhere.

Reply:
It's just a matter of replacing `g_advancedSettings.GetXXXExtensions` with a call to another singleton. If that require too much refactoring, just have `GetXXXExtensions` forward the calls as a temp solution; that's not an issue.

The important part is to start replacing these ugly mutable globals with things that have a clear lifetime.

Reply:
Sure. I'm a bit lazy, because my spare time is limited ;)